packages/salesforce-adapter/src/filters/formula_utils/parse.ts

```diff
@@ -0,0 +1,213 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { logger } from '@salto-io/logging'
+import {
+  isCPQRelationship, isCustom, isCustomLabel, isCustomMetadata, isCustomSetting, isObjectType, isParent, isParentField,
+  isProcessBuilderPrefix, isRelationshipField, isSpecialPrefix, isStandardRelationship, isUserField,
+} from './grammar'
+import {
+  createApiName, getField, getObject, parts, removeFirstAndLastChars, removePrefix, replaceRwithC, transformToId,
+  transformToUserField,
+} from './utils'
+import { mapCPQField } from './cpq'
+
+const log = logger(module)
+
+export class IdentifierType {
+  static CUSTOM_FIELD = new IdentifierType('customFields')
+  static STANDARD_FIELD = new IdentifierType('standardFields')
+  static CUSTOM_OBJECT = new IdentifierType('customObjects')
+  static STANDARD_OBJECT = new IdentifierType('standardObjects')
+  static CUSTOM_LABEL = new IdentifierType('customLabels')
+  static CUSTOM_SETTING = new IdentifierType('customSettings')
+  static CUSTOM_METADATA_TYPE_RECORD = new IdentifierType('customMetadataTypeRecords')
+  static CUSTOM_METADATA_TYPE = new IdentifierType('customMetadataTypes')
+  static UNKNOWN_RELATIONSHIP = new IdentifierType('unknownRelationships')
+
+  name: string
+
+  constructor(name: string) {
+    this.name = name
+  }
+}
+
+export type FormulaIdentifierInfo = {
+  type: IdentifierType
+  instance: string
+}
+
+export const parseField = (value: string, object: string | undefined = undefined): FormulaIdentifierInfo => {
```

Comment:
This is the same as this.
```suggestion
export const parseField = (value: string, object?: string): FormulaIdentifierInfo => {
```

Reply:
Ugh.